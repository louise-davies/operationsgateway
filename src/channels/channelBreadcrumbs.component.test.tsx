import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ChannelBreadcrumbs from './channelBreadcrumbs.component';

describe('Channel Breadcrumbs', () => {
  let currNode = '';
  const setCurrNode = vi.fn();
  const createView = () => {
    return render(
      <ChannelBreadcrumbs currNode={currNode} setCurrNode={setCurrNode} />
    );
  };

  it('should render correctly for path', () => {
    currNode = '/test/path';
    const view = createView();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly for root', () => {
    currNode = '/';
    const view = createView();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should call setCurrNode when link breadcrumb is clicked', async () => {
    const user = userEvent.setup();
    currNode = '/test/path';
    createView();

    await user.click(screen.getByRole('link', { name: 'test' }));
    expect(setCurrNode).toHaveBeenCalledWith('/test');
  });
});
