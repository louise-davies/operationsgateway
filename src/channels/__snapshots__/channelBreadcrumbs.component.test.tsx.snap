// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel Breadcrumbs should render correctly for path 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
          href="#"
        >
          All Channels
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="NavigateNextIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
          href="#"
        >
          test
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="NavigateNextIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
        >
          path
        </p>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`Channel Breadcrumbs should render correctly for root 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
        >
          All Channels
        </p>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
